src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala

```diff
@@ -111,6 +123,20 @@ private[xml] object StaxXmlParser extends Serializable {
     }
   }
 
+  def parseColumn(xml: String,
+                  schema: StructType,
+                  options: XmlOptions): Row = {
+    val factory: XMLInputFactory = XMLInputFactory.newInstance()
```

Comment:
Currently, 

1. XML string
2. XMLInputFactory instance is created
3. XML is parsed to `Row`

I think this can be avoided by:

```scala
class XmlDataToCatalyst(..) {
   @transient
  lazy val factory: XMLInputFactory = XMLInputFactory.newInstance()

  def nullSafeEval(...) = {
    factory.createXMLEventReader(new StringReader(xml))
  }
}
```

In this way, we will create one factory first, and then reuse it for every string input. (BTW, it's quite critical to fix this.).